pkg/deployment/kube_service.go

```diff
@@ -67,6 +85,59 @@ func CreateKubeServices(
 	return nil
 }
 
+func getAddressFromIPSet(item *dataplanev1.NodeSection,
+	namespacedName *types.NamespacedName,
+	helper *helper.Helper,
+) (string, discoveryv1.AddressType) {
+	// we go search for an IPSet
+	ipset := &infranetworkv1.IPSet{}
+	err := helper.GetClient().Get(context.Background(), *namespacedName, ipset)
+	if err != nil {
+		// No IPsets found, lets try to get the HostName as last resource
+		if isValidDomain(item.HostName) {
+			return item.HostName, discoveryv1.AddressTypeFQDN
+		}
+		// No IP address or valid hostname found anywhere
+		helper.GetLogger().Info("Did not found a valid hostname or IP address")
+		return "", ""
+	}
+	// check that the reservations list is not empty
+	if len(ipset.Status.Reservation) > 0 {
+		// use the first IP in the reservations list. This is CtlPlane network.
+		return ipset.Status.Reservation[0].Address, discoveryv1.AddressTypeIPv4
```

Comment:
Is CtlPlane always the first reservation? And is that always the right IP to use, even when using network isolation?

Reply:
I dont have the answer to those questions, only more questions. 

If it is not the first reservation, we will get whatever is the first reservation, as this is not exactly important for the Service itself.

The second question is more of a functional one and impacts the previous one. I don't really know if using the CtlPlane connection is an issue or not. We can use the IP we want as long as it is accessible from the ControlPlane, the CtlPlane seemed the most reasonable, but at this point (until I get more information), it is largely irrelevant.

Maybe we can add a new parameter on the KubeService API so that the user itself is able to define the network the Service should use. Would that be reasonable?

Reply:
Yes..I think adding network to KubeService makes sense. If not set, default to CtlPlane.